docs/manual/tsort.md

```diff
@@ -2,160 +2,56 @@
-```
-    "prep",           RPMSENSE_SCRIPT_PREP
-    "build",          RPMSENSE_SCRIPT_BUILD
-    "install",        RPMSENSE_SCRIPT_INSTALL
-    "clean",          RPMSENSE_SCRIPT_CLEAN
-```
-but let's not go there (yet).
-
-For giggles, you can also do stuff like
-```
-	Requires(pre,post): /bin/sh
-```
-
-By marking dependencies more precisely, rpm can distinguish between
-an upgrade context (like the use of grep in %post above) and an installed
-context (like the autogenerated Requires: in a package that includes a
-script with #!/bin/sh), and that permits rpm to differentiate pre-requisites
-from co-requisites while doing package ordering.
-
-Here's what cures the libtermcap <-> bash loop:
-```
-	Requires(postun): /bin/sh
-```
-which, since the dependency is clearly not useful or necessary in determining
-install ordering, is safely ignored.
-
-## Side Effects
-
-One of the side effects of changing the package install ordering, is that
-there are a handful of new loops that are detected. Here's what I found
-looking at supported Red Hat releases:
-
-```
-    ghostscript-fonts	ghostscript
-    /* 7.0 only */
-    pango-gtkbeta-devel	pango-gtkbeta
-    XFree86		Mesa
-    compat-glibc	db2
-    compat-glibc	db1
-    pam			initscripts
-    kernel		initscripts
-    initscripts		sysklogd
-    /* 6.2 */
-    egcs-c++		libstdc++
-    /* 6.1 */
-    pilot-link-devel	pilot-link
-    /* 5.2 */
-    pam			pamconfig
-```
-
-Why are there new loops? Because tsort is trying to use all of the
-dependency relations for ordering, while the previous tsort ignored all
-Requires: from added packages.
-
-Except for the "well known" libtermcap <-> bash loop (which is just wrong),
-all of the other dependencies are simply not needed in an upgrade context
-to perform package ordering. Please note that all of the known to cause
-loop dependencies listed above are, for now, explicitly ignored when
-determining package install ordering.
-
-## Summary
-
-So what does this all mean? Basically not much, unless you find yourself
-trying to specify dependencies amongst a set of packages correctly and
-happen to create a dependency loop.
-
-And, before you start adding the new-fangled syntax to packages, please
-remember that rpm will almost certainly be auto-generating fine-grained
-dependencies for %post et al scriptlets pretty soon. Truly, rpm needs to
-make packaging easier, not provide Yet More Complicated Syntax in spec files.
-
-With thanks to Ken Estes for doing the implementation in bash2 that makes
-it possible to auto-generate scriptlet dependencies, blame me for the long,
-slow deployment.
+# Package ordering
+
+Rpm orders transactions solely based on package dependency information:
+dependencies first on install, last on erase. If the install/erase
+order is not right, then either the dependency information is incomplete,
+incorrect or there are unresolvable dependency loops.
+
+`Requires` are the primary data for ordering. If package A Requires B,
```

Comment:
Not sure if "primary data" makes a lot of sense here. May be the match between a Dependency like Requires and a Provide or Name of another should be mentioned here instead. This would be the place to establish this  as a "Dependency" so the word has a defined meaning for the rest of the document.